packages/core/src/scripts/migrate-oz-cli-project.ts

```diff
@@ -0,0 +1,426 @@
+  };
+}
+
+function transformStorageItem(storageItem: LegacyStorageItem): StorageItem {
+  return {
+    contract: storageItem.contract,
+    label: storageItem.label,
+    type: transformTypeName(storageItem.type),
+    // TODO reconstruct path and line if sourcecode is available
+    src: storageItem.path,
+  };
+}
+
+function transformTypes(oldTypes: LegacyTypes): Record<string, TypeItem> {
+  const newTypes: Record<string, TypeItem> = {};
+  for (const typeName in oldTypes) {
+    newTypes[transformTypeName(typeName)] = transformType(getTypeKind(typeName), oldTypes[typeName]);
+  }
+  return newTypes;
+}
+
+function transformType(typeKind: string, oldType: LegacyType): TypeItem {
+  switch (typeKind) {
+    case 'Struct':
+      return {
+        label: stripContractName(oldType.label),
+        members: (oldType.members as StructMember[]).map(member => ({
+          label: stripContractName(member.label),
+          type: transformTypeName(member.type),
+        })),
+      };
+    case 'Enum':
+      return {
+        label: stripContractName(oldType.label),
+        members: oldType.members,
+      };
+    default:
+      return {
+        label: oldType.label,
+      };
+  }
+}
+
+function transformTypeName(typeName: string): string {
+  switch (getTypeKind(typeName)) {
+    case 'Struct':
+      return transformStructTypeName(typeName);
+    case 'Enum':
+      return transformEnumTypeName(typeName);
+    case 'Mapping':
+      return transformMappingTypeName(typeName);
+    case 'DynArray':
+      return transformDynArrayTypeName(typeName);
+    case 'StaticArray':
+      return transformStaticArrayTypeName(typeName);
+    case 'Elementary':
+      return typeName;
+    default:
+      throw new Error(`Unknown type: ${typeName}`);
+  }
+}
+
+function transformStructTypeName(typeName: string): string {
+  const valueType = stripContractName(getValueType(typeName));
+  return `t_struct(${valueType})_storage`;
+}
+
+function transformEnumTypeName(typeName: string): string {
+  const valueType = stripContractName(getValueType(typeName));
+  return `t_enum(${valueType})`;
+}
+
+function transformMappingTypeName(typeName: string): string {
+  const valueType = transformTypeName(getValueType(typeName));
+  return `t_mapping(${valueType})`;
+}
+
+function transformDynArrayTypeName(typeName: string): string {
+  const valueType = transformTypeName(getValueType(typeName));
+  return `t_array(${valueType})dyn_storage`;
+}
+
+function transformStaticArrayTypeName(typeName: string): string {
+  const size = optimisticMatch(typeName, /:(\d+)/)[1];
+  const valueType = transformTypeName(getValueType(typeName));
+  return `t_array(${valueType})${size}_storage`;
+}
+
+function getValueType(typeName: string): string {
+  return optimisticMatch(typeName, /<(.+)>/)[1];
+}
+
+function stripContractName(s: string): string {
+  return optimisticMatch(s, /(.+)\.(.+)/)[2];
+}
+
+function optimisticMatch(s: string, rg: RegExp): string[] {
```

Comment:
Can you remove this function and use the non-null assertion operator `!`?

e.g. `s.match(/.../)![2]`

Reply:
No, `optimisticMatch` returns an array with the same element if no match is found.

Reply:
But that isn't used anywhere.

Reply:
Yes, in `stripContractName` for example